src/lib/users/ssh_authorized_keyring.rb

```diff
@@ -0,0 +1,213 @@
+      private
+
+      # @return [String] Relative path to the SSH directory inside users' home
+      SSH_DIR = ".ssh".freeze
+      # @return [String] Authorized keys file name
+      AUTHORIZED_KEYS_FILE = "authorized_keys".freeze
+      # @return [String] Permissions to be set on SSH_DIR directory
+      SSH_DIR_PERMS = "0700".freeze
+      # @return [String] Permissions to be set on `authorized_keys` file
+      AUTHORIZED_KEYS_PERMS = "0600".freeze
+
+      # Determine the path to the user's SSH directory
+      #
+      # @param home [String] Home directory
+      # @return [String] Path to the user's SSH directory
+      #
+      # @see SSH_DIR
+      def ssh_dir_path(home)
+        File.join(home, SSH_DIR)
+      end
+
+      # Determine the path to the user's authorized keys file
+      #
+      # @param home [String] Home directory
+      # @return [String] Path to authorized keys file
+      #
+      # @see SSH_DIR
+      # @see AUTHORIZED_KEYS_FILE
+      #
+      # @see #ssh_dir_path
+      def authorized_keys_path(home)
+        File.join(ssh_dir_path(home), AUTHORIZED_KEYS_FILE)
+      end
+
+      # Find or creates the SSH directory
+      #
+      # This method sets up the SSH directory (usually .ssh). Although only 1
+      # level is needed (as SSH directory lives under $HOME/.ssh), this code
+      # should support changing SSH_DIR to something like `.config/ssh`.
+      #
+      # @param home  [String] Home directory where SSH directory must be created
+      # @param owner [Fixnum] Owner's UID
+      # @param group [Fixnum] Owner's GID
+      # @param perms [String] Permissions (in form "0700")
+      # @return [String] Returns the path to the first created directory
+      def create_ssh_dir(home, owner, group, perms)
+        ssh_dir = ssh_dir_path(home)
+        return ssh_dir if FileUtils::Exists(ssh_dir)
+        ret = Yast::SCR.Execute(Yast::Path.new(".target.mkdir"), ssh_dir)
+        log.info("Creating SSH directory: #{ret}")
+        return false unless ret
+        set_owner_and_perms(ssh_dir, owner, group, perms)
+      end
+
+      # Write authorized keys file
+      #
+      # @param path  [String] Path to file/directory
+      # @param owner [Fixnum] Owner's UID
+      # @param group [Fixnum] Owner's GID
+      # @param perms [String] Permissions (in form "0700")
+      def write_file(home, owner, group, perms)
+        path = authorized_keys_path(home)
+        file = Yast::Users::SSHAuthorizedKeysFile.new(path)
+        file.keys = keys[home]
+        log.info "Writing #{keys[home].size} keys in #{path}"
+        file.save && set_owner_and_perms(path, owner, group, perms)
+      end
+
+      # Set owner and permissions for a given directory/file
+      #
+      # @param path  [String] Path to file/directory
+      # @param owner [Fixnum] Owner's UID
+      # @param group [Fixnum] Owner's GID
+      # @param perms [String] Permissions (in form "0700")
+      def set_owner_and_perms(path, owner, group, perms)
```

Comment:
Oh... then you could use [Yast::Execute.on_target](https://github.com/yast/yast-yast2/blob/master/library/system/src/lib/yast2/execute.rb#L41).

IMHO we should avoid using SCR unless really needed, personally I consider it as another "YCP zombie"...